images/metalk8s-keepalived/generate-config.py

```diff
@@ -14,6 +15,12 @@
 
 
 def get_interface_from_ip(ip):
+    # Check if the IP does not already sit on an interface (if yes then use it)
+    for interface in netifaces.interfaces():
+        for link in netifaces.ifaddresses(interface).get(netifaces.AF_INET) or []:
+            if link.get("addr") == ip:
+                return interface
+
     # NOTE: We do not have any easy way (without an external lib)
```

Comment:
This is the logic to handle the "loopback" bug, we see if there is an interface that already has the VIP.

AFAIK, this lib cannot be used to get the route.

The logic is if the VIP already sits on an interface (it usually shouldn't happen but it could if for whatever reason keepavlied get killed) we keep it, otherwise we "search" it based on the routes